scico/linop/radon_svmbir.py

```diff
@@ -30,39 +30,76 @@
+class TomographicProjector(LinearOperator):
+    r"""Tomographic projector based on svmbir.
 
     Perform tomographic projection of an image at specified angles, using
     the `svmbir <https://github.com/cabouman/svmbir>`_ package. The
-    ``is_masked`` option selects whether a valid region for projections
+    `is_masked` option selects whether a valid region for projections
     (pixels outside this region are ignored when performing the
     projection) is active. This region of validity is also respected by
     :meth:`.SVMBIRSquaredL2Loss.prox` when :class:`.SVMBIRSquaredL2Loss`
-    is initialized with a :class:`ParallelBeamProjector` with this option
+    is initialized with a :class:`TomographicProjector` with this option
     enabled.
+
+    A brief description of the supported scanner geometries can be found
+    in the `svmbir documentation <https://svmbir.readthedocs.io/en/latest/overview.html>`_.
+    Both parallel and fan beam geometries are supported.
+
+    .. list-table::
+
+       * - .. figure:: https://svmbir.readthedocs.io/en/latest/_images/geom-parallel.jpg
+
+              Fig 1. Parallel beam geometry.
+
+         - .. figure:: https://svmbir.readthedocs.io/en/latest/_images/geom-fan.jpg
+
+              Fig 2. Fan beam geometry.
     """
 
     def __init__(
         self,
         input_shape: Shape,
         angles: np.ndarray,
         num_channels: int,
-        center_offset: Optional[float] = 0.0,
-        is_masked: Optional[bool] = False,
+        center_offset: float = 0.0,
+        is_masked: bool = False,
+        geometry: str = "parallel",
+        dist_source_detector: Optional[float] = None,
+        magnification: Optional[float] = None,
     ):
         """
         Args:
-            input_shape: Shape of the input array.
+            input_shape: Shape of the input array. May be of length 2 (a
+                2D array) or 3 (a 3D array). When specifying a 2D array,
+                the format for the input_shape is `(num_rows, num_cols)`.
+                For a 3D array, the format for the input_shape is
+                `(num_slices, num_rows, num_cols)`, where `num_slices`
+                denotes the number of slices in the input, and `num_rows`
+                and `num_cols` denote the number of rows and columns in a
+                single slice of the input. (A slice is a plane
+                perpendicular to the axis of rotation of the tomographic
+                system. Note that `input_shape=(num_rows, num_cols)` and
+                `input_shape=(1, num_rows, num_cols)` result in the
+                same underlying projector.
             angles: Array of projection angles in radians, should be
                 increasing.
-            num_channels: Number of pixels in the sinogram.
+            num_channels: Number of detector channels in the sinogram
+                data.
             center_offset: Position of the detector center relative to
```

Comment:
It would help to explain where the center of rotation is.

Reply:
Added some extra explanation on the center of rotation. Did you mean something different?

Reply:
Looks good.